shared/util/second-timer.js

```diff
@@ -0,0 +1,52 @@
+// @flow
+
+export opaque type TickerID: number = number
+
+// Global ID for refs
+let id: TickerID = 0
+
+type Ref = {
+  id: TickerID,
+  fn: () => void,
+}
+
+class Ticker {
+  refs: Array<Ref> = []
+  intervalID: ?IntervalID
+
+  addObserver = (fn: () => void): TickerID => {
+    if (!this.intervalID) {
+      this.intervalID = setInterval(this.loop, 1000)
+    }
+    id++
+    this.refs.push({fn, id})
+    return id
+  }
+
+  removeObserver = (id: TickerID) => {
+    const index = this.refs.findIndex(r => r.id === id)
+    if (index >= 0) {
+      this.refs.splice(index, 1)
+      if (this.refs.length === 0 && this.intervalID) {
+        clearInterval(this.intervalID)
+        this.intervalID = null
+      }
+      return true
+    }
+    return false
+  }
+
+  loop = () => {
+    this.refs.forEach(r => setImmediate(r.fn))
+  }
+}
```

Comment:
you might want to add a localDebug flag like the outstandingRPC one where we can log if there's outstanding listeners to this thing